### PR TITLE
Update `filecoin-cxx` deps, mainly `multihash`

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
This change is a first pass in getting the dependecies for `filecoin-cxx` updated, to eventually be able to drop the unmaintained crate `proc-macro-error`.

This change updates `multihash` to `v0.16.3`->`v0.18.1`. In a follow up PR, the goal will be to deploy `v0.19.1`, which drops the unwanted crate as a transient dependency.

Resolves https://github.com/brave/brave-browser/issues/40947

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by antonok-edm — APPROVED** [review]

**Review by cypt4 — APPROVED** [review]

File: third_party/rust/chromium_crates_io/Cargo.lock [hunk 1]
```diff
@@ -175,6 +175,14 @@ name = "bitflags"
 version = "1.3.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
+[[package]]
```
Comment (bridiver) [inline third_party/rust/chromium_crates_io/Cargo.lock:178]: bitflags 2.6.0 already exists in upstream

File: third_party/rust/bitflags/v2/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# @generated from third_party/rust/chromium_crates_io/BUILD.gn.hbs by
+# tools/crates/gnrt.
+# Do not edit!
+
+import("//build/rust/cargo_crate.gni")
+
+cargo_crate("lib") {
```
Comment (bridiver) [inline third_party/rust/bitflags/v2/BUILD.gn:12]: bitflags 2.6.0 already exists in upstream

Comment (cdesouza-chromium) [reply third_party/rust/bitflags/v2/BUILD.gn]: Done.

File: third_party/rust/chromium_crates_io/gnrt_config.toml [hunk 3]
```diff
@@ -280,7 +280,7 @@ extra_build_script_src_roots = [ "entities.rs" ]
 build_script_outputs = [ "generated.rs", "named_entities.rs", ]
 
 [crate.multihash]
-extra_kv = { rustenv = [ "CARGO_MANIFEST_DIR=\" + rebase_path(\"//brave/third_party/rust/chromium_crates_io/vendor/multihash-0.16.3\") + \"" ]}
+extra_kv = { rustenv = [ "CARGO_MANIFEST_DIR=\" + rebase_path(\"//brave/third_party/rust/chromium_crates_io/vendor/multihash-0.18.1\") + \"" ]}
```
Comment (bridiver) [inline third_party/rust/chromium_crates_io/gnrt_config.toml:283]: I'm not sure we still need this, looks like it's added in run_build_script.py now

Comment (cdesouza-chromium) [reply third_party/rust/chromium_crates_io/gnrt_config.toml]: Done

**Review by bridiver — APPROVED** [review]

File: third_party/rust/chromium_crates_io/gnrt_config.toml (hunk 3 above)
Comment (cdesouza-chromium) [reply third_party/rust/chromium_crates_io/gnrt_config.toml]: Removing this line seems to trigger the following failure

```
error: `CARGO_MANIFEST_DIR` env variable not set.
  --> ../../brave/third_party/rust/chromium_crates_io/vendor/multihash-0.18.1/src/multihash_impl.rs:11:34
   |
11 |   #[derive(Copy, Clone, Debug, Eq, Multihash, PartialEq)]
   |                                    ^^^^^^^^^ in this derive macro expansion
  --> ./../../brave/third_party/rust/chromium_crates_io/vendor/synstructure-0.12.6/src/macros.rs:94:9
  ::: ./../../brave/third_party/rust/chromium_crates_io/vendor/synstructure-0.12.6/src/macros.rs:96:40
   |
   = note: in this expansion of `#[derive(Multihash)]`

error: aborting due to 1 previous error
```

I'm not quite sure why `run_build_script.py` is failing to set `CARGO_MANIFEST_DIR` for `multihash` in particular, but this failure seems to be the reason this line has been here in the first place.

**Pull request merged**